### PR TITLE
Add modal to edit environment strategy

### PR DESCRIPTION
**Pull request by rottencandy**
## Fixes 
https://issues.redhat.com/browse/HAC-2004

## Description
Add modal to edit environment strategy.

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

https://user-images.githubusercontent.com/20013884/184175603-2481905b-33f4-4c9e-ab38-6a5f45cdd426.mp4

## Browser conformance: 

- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by rottencandy** [comment]
cc @openshift/team-hacdev-ux

**Comment by Ranelim** [comment]
Thank you @rottencandy 
is this modal part of App Studio? I'm not sure it was a HACBS deliverable.

@MariaLeonova Do you know more about this?

**Comment by itsptk** [comment]
@Ranelim This was designed by @bmignano for app studio via: https://issues.redhat.com/browse/DTUX-1278

https://docs.google.com/document/d/1OG0eSYTOAbXEGFW4HaZYkImX_K0nNkQfTZXCD5F267I/edit?usp=sharing

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Comment by rottencandy** [comment]
screenshot-1660736543

File: src/components/Environment/EditStrategyModal.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import { Formik, FormikValues } from 'formik';
+import { EnvironmentModel } from '../../models';
+import { CheckboxField, DropdownField } from '../../shared';
+import { EnvironmentKind } from '../../types';
+import { ComponentProps, createModalLauncher } from '../modal/createModalLauncher';
+
+const dropdownItems = [
+  { value: 'Automatic', key: 'AppStudioAutomated' },
+  { value: 'Manual', key: 'Manual' },
+];
```
Comment (sahil143) [inline src/components/Environment/EditStrategyModal.tsx:26]:
```suggestion
const dropdownItems = Object.entries(EnvironmentDeploymentStrategy).map(([value, key]) => ({key, value}));
```

File: src/components/Environment/environment-actions.ts [hunk 2]
```diff
@@ -18,5 +19,15 @@ export const useEnvironmentActions = (environment: EnvironmentKind): Action[] =>
...
+    {
+      cta: () =>
+        showModal(
+          createEditStrategyModal({
+            obj: environment,
+          }),
+        ),
+      id: `edit-strategy-${environment.metadata.name.toLowerCase()}`,
+      label: 'Edit deployment strategy',
+    },
```
Comment (sahil143) [inline src/components/Environment/environment-actions.ts:31]: this action should be before delete action.

**Review by sahil143 — CHANGES_REQUESTED** [review]

File: src/components/Environment/environment-actions.ts (hunk 2 above)
Comment (rottencandy) [reply src/components/Environment/environment-actions.ts]: Updated.

**Review by rohitkrai03 — APPROVED** [review]
/lgtm

**Pull request merged**